hledger-lib/Hledger/Data/Amount.hs

```diff
@@ -580,56 +579,64 @@ canonicaliseAmount styles a@Amount{acommodity=c, astyle=s} = a{astyle=s'}
 
 instance Semigroup MixedAmount where
   (<>) = maPlus
+  sconcat = maSum
+  stimes n = multiplyMixedAmount (fromIntegral n)
 
 instance Monoid MixedAmount where
   mempty = nullmixedamt
+  mconcat = maSum
```

Comment:
Just curious, why do you implement more instance methods now ? They were using default implementations before.. and now you've provided more efficient implementations ?

Reply:
Correct. In particular, this now does a strict sum, rather than a lazy sum.